mlir/include/mlir/Dialect/AMDGPU/Transforms/Passes.td

```diff
@@ -31,4 +31,20 @@ def AmdgpuEmulateAtomicsPass : Pass<"amdgpu-emulate-atomics"> {
                         "Chipset that these operations will run on">];
 }
 
+def AmdgpuResolveStridedMetadataPass : Pass<"amdgpu-resolve-strided-metadata"> {
+  let summary = "Resolve memref.extract_strided_metadata on AMDGPU ops";
+  let description = [{
+    This pass rrewrites `memref.extract_strided_metadata` operations
+    targeting the AMDGPU dialect casts.
+
+    It's mainly meant for testing - please incorporate the patterns into your
+    own extract-strided-metadata passes (or run memref's expand-strided-metadata
+    again after this).
```

Comment:
If it's a test pass could we remove it from any public headers and register it in `mlir-opt.cpp` manually?

Reply:
"mainly for testing" != "only for testing" - this is a valid pass to run on some code to clean up stray strided metadata extractions